### PR TITLE
chore: add ICS-20 docs

### PR DESCRIPTION
**Pull request by GAtom22**
closes https://linear.app/evmos/issue/ENG-1306/ics20-precompile-docs

### CODE REVIEW

**Review by fedekunze — COMMENTED** [review]
@GAtom22 seems that we also need to add it to the list of addresses and to `evm-extensions.md`

File: docs/develop/smart-contracts/evm-extensions/ibc-transfer.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+Find the [ABI in the evmos/extensions repo](https://github.com/evmos/extensions/blob/main/precompiles/abi/ibctransfer.json).
+
+## Transactions
+
+- `approve`
+
+    ```solidity
+    /// @dev Approves IBC transfer with a specific amount of tokens.
+    /// @param spender spender The address which will spend the funds.
+    /// @param allocations the allocations for the authorization.
```
Comment (MalteHerrmann) [inline docs/develop/smart-contracts/evm-extensions/ibc-transfer.md:29]:
```suggestion
    /// @param allocations The allocations for the authorization.
```

File: docs/develop/smart-contracts/evm-extensions/ibc-transfer.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        address spender,
+        Allocation[] calldata allocations
+    ) external returns (bool approved);
+    ```
+
+- `revoke`
+
+    ```solidity
+    /// @dev Revokes IBC transfer authorization for a specific spender
+    /// @param spender spender The address which will spend the funds.
```
Comment (MalteHerrmann) [inline docs/develop/smart-contracts/evm-extensions/ibc-transfer.md:40]: if this is also in the codebase we should adjusted it accordingly
```suggestion
    /// @param spender The address which will spend the funds.
```

**Review by MalteHerrmann — APPROVED** [review]
looks good, pending mentioning it on the EVM Extensions landing page and in the extensions addresses table

**Review by fedekunze — APPROVED** [review]

**Pull request merged**